src/sds.c

```diff
@@ -192,6 +192,25 @@ sds sdsdup(const sds s) {
     return sdsnewlen(s, sdslen(s));
 }
 
+/*
+ * This method returns the minimum amount of bytes required to store the sds (header + data + NULL terminator).
+ */
+static inline size_t sdsminlen(sds s) {
+    return sdslen(s) + sdsHdrSize(s[-1]) + 1;
+}
+
+/* This method copies the sds `s` into `buf` which is the target character buffer. */
+size_t sdscopytobuffer(unsigned char *buf, size_t buf_len, sds s, uint8_t *hdr_size) {
+    size_t reqd_keylen = sdsminlen(s);
+    if (buf == NULL) {
+        return reqd_keylen;
+    }
+    assert(buf_len >= reqd_keylen);
+    memcpy(buf, sdsAllocPtr(s), reqd_keylen);
```

Comment:
```suggestion
    size_t required_keylen = sdsminlen(s);
    if (buf == NULL) {
        return required_keylen;
    }
    assert(buf_len >= required_keylen);
    memcpy(buf, sdsAllocPtr(s), required_keylen);
```
I don't think we should use abbreviations unless they are consistently used.